pkg/csi/service/wcpguest/controller.go

```diff
@@ -1345,11 +1355,27 @@ func (c *controller) ListVolumes(ctx context.Context, req *csi.ListVolumesReques
 
 func (c *controller) GetCapacity(ctx context.Context, req *csi.GetCapacityRequest) (
 	*csi.GetCapacityResponse, error) {
-
 	ctx = logger.NewContextWithLogger(ctx)
 	log := logger.GetLogger(ctx)
 	log.Infof("GetCapacity: called with args %+v", *req)
-	return nil, status.Error(codes.Unimplemented, "")
+
+	var totalcapacity, maxvolumesize int64
+	zonesMap := commonco.ContainerOrchestratorUtility.GetZonesForNamespace(c.supervisorNamespace)
+	if req.AccessibleTopology != nil {
+		segments := req.AccessibleTopology.GetSegments()
+		if _, exists := zonesMap[segments["topology.csi.vmware.com/k8s-zone"]]; !exists {
+			totalcapacity = 0
+			maxvolumesize = 0
+		} else {
+			totalcapacity = math.MaxInt64
+			maxvolumesize = math.MaxInt64
+		}
+	}
+	resp := &csi.GetCapacityResponse{
+		AvailableCapacity: totalcapacity,
```

Comment:
In case of non-topology aware setup, should we return math.MaxInt64 as total capacity & maxvolumesize ?

Reply:
topology is enabled by default with latest PVCSI provisioner.

Reply:
when FSS is disabled we can return `return nil, status.Error(codes.Unimplemented, "")`